tests/sampling/test_jax.py

```diff
@@ -503,3 +509,31 @@ def test_convergence_warnings(caplog, nuts_sampler):
 
     [record] = caplog.records
     assert re.match(r"There were \d+ divergences after tuning", record.message)
+
+
+@pytest.mark.parametrize("method", ["advi", "fullrank_advi"])
+def test_vi_sampling_jax(method):
+    with pm.Model() as model:
+        x = pm.Normal("x")
+        pm.fit(10, method=method, fn_kwargs=dict(mode="JAX"))
```

Comment:
To be consistent with `pm.sample` and the `nuts_sampler=` arg, should we have a dedicated argument for the VI backend instead of kwargs?

Reply:
I vote yes, this API looks super weird.

Reply:
What looks weird? This is the compilation mode, would be exactly the same if you wanted to use Numba or JAX for the PyMC nuts sampler or for prior/posterior predictive.

The only thing I would change is the name of fn_kwargs, which is called compile_kwargs I think in those other functions

Reply:
Wouldn't this be what the user would have to do if they wanted to run VI on JAX?

Reply:
I don't understand the question, this PR is just doing minor tweaks so the PyMC VI module can compile to JAX. It's not linking to specific JAX VI libraries.

Reply:
You call it "VI module can compile to JAX", I call it "This PR adds an exciting feature that allows users to run VI on the much faster JAX backend wich also supports GPUs. Simply call `pm.fit(backend="jax")` to speed up your VI."

Reply:
I'm just saying this is not specific to VI nor JAX. Any function that compiles a Pytensor function allows JAX or Numba mode: draw, sample, prior/posterior/predictive, compile_logp...

I don't see why it should be a VI only argument. And my comment above is this is not equivalent to `nuts_sampler`.

If we want to add a backend kwarg as a shortcut for mode we would add everywhere for consistency.

Reply:
> If we want to add a backend kwarg as a shortcut for mode we would add everywhere for consistency.

Where else would you add it?

Reply:
All the functions mentioned above that already have a `compile_kwargs` equivalent to this `fn_kwargs` in VI. Neither kwarg is new.

Literally any function that compiles a Pytensor function should have the same kwargs API

Reply:
https://github.com/pymc-devs/pymc/blob/d8816c711df65974fe9785a877af1fccef2a1e14/pymc/sampling/forward.py#L368

Reply:
I'm fine adding it anywhere for consistency, but I mainly care about having it where a non-expert user would actually use it, like here.

The one other relevant case I can think of it would actually almost be detrimental: `pm.sample(backend="JAX")` where as a user I would expect it just run everything on JAX, which it wouldn't do because the sampler is still in Python (unless we made it call `nuts_sampler="numpyro"` under the hood).

I don't think the users should need to know about these finer points, they should just get a "make it fast with jax" button in the right place.

Reply:
We used this for sample_posterior_predictive for projects just last week, as we were sampling new variables that had heavy matmuls, went down from hours to minutes.

pm.sample is still useful as you can sample discrete variables with JAX this way.

Reply:
This is still doing python loops, it's exactly the same argument you need to do for pm.sample.

It's different than linking to a JAX VI library, which would be equivalent to the `nuts_sampler` kwarg that Chris mentioned in the first comment

Reply:
> This is still doing python loops, it's exactly the same argument you need for pm.sample.

Oh, I somehow assumed that VI was implemented mostly in PyTensor?

Reply:
As for this, I'd prefer to focus this PR on backend compatibility and later address possible API changes in a new issue + PR. Agreed that there is inconsistency, we need to resolve that, but this will only defer the push to main with at least some working solution which went through many issues already.

Reply:
Agreed @ferrine. My only suggestion is to switch `fn_kwargs` to `compile_kwargs` which we use in the other sample methods